keyboards/matrix/noah/matrix.c

```diff
@@ -0,0 +1,166 @@
+/**
```

Comment:
Is there a specific reason for the custom matrix?

Reply:
I tried the current matrix implement for stm32 several weeks ago, but they can't work properly on stm32f4. I did not investigate the root cause, and because modified the codes on qmk parts will introduced massive verification process on other hardware(stm32f3, stm32f1 and stm32f0). So custom matrix was used to avoid to modify the qmk matrix codes.

Reply:
Okay.  My guess, it may be the `config_common.h` file, or something in the matrix.  But yeah, that's understandable.

Though, really, I'd prefer to see the F4 stuff in the core, eventually, the way that a lot of the F0, F1 and F3 stuff is. But that's outside the scope of this PR.